SwrveSDK/SwrveClient.brs

```diff
@@ -1,1112 +1,1022 @@
-    m.global.unobserveField("SwrveGlobalIdentifyExternalIDCallback")
-    m.global.unobserveField("SwrveGetNewResourcesDiff")
-    m.global.unobserveField("SwrveGlobalCurrencyGiven")
-    m.global.unobserveField("SwrveGlobalUserUpdateWithDate")
-    m.global.unobserveField("SwrveGlobalIAPWithoutReceipt")
-    m.global.unobserveField("SwrveGlobalFlushAndClean")
-
-    m.global.removeField("SwrveDebug")
-    m.global.removeField("SwrveIsQAUser")
-    m.global.removeField("SwrveCustomCallback")
-    m.global.removeField("swrveShowIAM")
-    m.global.removeField("swrveCurrentIAM")
-    m.global.removeField("SwrveShutdown")
-    m.global.removeField("swrveResourcesAndCampaigns")
-    m.global.removeField("SwrveResourcesDiffObjectReady")
-    m.global.removeField("swrveAssetsReady")
-    m.global.removeField("SwrveEvent")
-    m.global.removeField("SwrveClickEvent")
-    m.global.removeField("SwrvePurchaseEvent")
-    m.global.removeField("SwrveUserUpdate")
-    m.global.removeField("SwrveImpressionEvent")
-    m.global.removeField("messageWillRender")
-    m.global.removeField("swrveSDKHasCustomRenderer")
-    m.global.removeField("SwrveGlobalIdentifyExternalID")
-    m.global.removeField("SwrveGlobalIdentifyExternalIDCallback")
-    m.global.removeField("SwrveGetNewResourcesDiff")
-    m.global.removeField("SwrveGlobalCurrencyGiven")
-    m.global.removeField("SwrveGlobalUserUpdateWithDate")
-    m.global.removeField("SwrveGlobalIAPWithoutReceipt")
-    m.global.removeField("SwrveGlobalFlushAndClean")
+	m.swrve_config.stopped = false
+	if m.delayTimer <> Invalid then m.delayTimer.control = "start"
+end function
+
+function SwrveOnShutdown() as Object
+	if(m.delayTimer <> Invalid)
+		m.delayTimer.control = "stop"
+		m.delayTimer = Invalid
+	end if
+
+	SWLogWarn("Shutdown initiated.")
+	SWLogWarn("Shutdown initiated....Flushing queue")
+
+	SWLogWarn("Shutdown initiated....clearing persistent storage")
+	SwrveClearKeyFromPersistence(SwrveConstants().SWRVE_LAST_SESSION_DATE_KEY)
+	SwrveClearKeyFromPersistence(SwrveConstants().SWRVE_START_SESSION_DATE_KEY)
+	SwrveClearKeyFromPersistence(SwrveConstants().SWRVE_SEQNUM)
+	SwrveClearKeyFromPersistence(SwrveConstants().SWRVE_USER_ID_KEY)
+	SwrveClearKeyFromPersistence(SwrveConstants().SWRVE_EVENTS_STORAGE)
+	SwrveClearKeyFromPersistence(SwrveConstants().SWRVE_ETAG_FILENAME)
+
+	SWLogWarn("Shutdown initiated....Clearing memory")
+	SWLogWarn("Shutdown. To use Swrve features you will need to reinitialize.")
+
+	m.startSession = Invalid
+	m.userCampaigns = Invalid
+	m.userResources = Invalid
+	m.installDate = Invalid
+	m.joinedDate = Invalid
+	m.eventsQueue = Invalid
+	m.eventsQAQueue = Invalid
+	'm.swrveNextFlush = Invalid
+	m.numberOfMessagesShown = Invalid
+	m.startSessionAsSeconds = Invalid
+	m.swrveNextUpdateCampaignsAndResources = Invalid
+
+	m.swrve_config = Invalid
+	SwrveNode = getSwrveNode()
+	fields = SwrveNode.keys()
+
+	' Stop watching for public api events
+	for each field in fields
```

Comment:
Since all the Swrve global fields are now scoped on the `SwrveNode` we can simply loop over and unobserved each field.